File: typescript/src/types/decision.ts
import { PolicyReference } from "./policy-reference.js";

import type { JsonValue } from "../types/Json.js";

/**
 * Parmana Trust Core
 *
 * Decision
 *
 * Immutable result produced by evaluating an
 * Intent against a Policy.
 *
 * Decision does not create authority.
 * Decision does not grant authorization.
 * Decision does not modify intent.
 *
 * Decision records only the outcome of
 * deterministic Policy evaluation.
 */
export interface Decision {
  /**
   * Unique Decision identifier.
   */
  readonly decisionId: string;

  /**
   * Intent evaluated by this Decision.
   */
  readonly intentId: string;

  /**
   * Exact Policy used.
   */
  readonly policy: PolicyReference;

  /**
   * Runtime signals evaluated by the policy.
   *
   * These are captured to support deterministic
   * replay and independent verification.
   */
  readonly signals: Record<string, JsonValue>;

  /**
   * Policy evaluation outcome.
   */
  readonly outcome: DecisionOutcome;

  /**
   * Optional explanation.
   */
  readonly reason?: string;

  /**
   * UTC timestamp when evaluation completed.
   */
  readonly evaluatedAt: Date;
}

/**
 * Canonical Decision outcomes.
 */
export enum DecisionOutcome {
  APPROVED = "APPROVED",

  REJECTED = "REJECTED",
}

File: python-old/parmana/models/decision.py
"""
Decision model.

Immutable result produced by deterministic policy evaluation.
"""

from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum
from typing import Any

from .policy_reference import PolicyReference


class DecisionOutcome(str, Enum):
    APPROVED = "APPROVED"
    REJECTED = "REJECTED"


@dataclass(frozen=True, slots=True)
class Decision:
    """
    Immutable policy evaluation result.
    """

    decision_id: str

    intent_id: str

    policy: PolicyReference

    signals: dict[str, Any] = field(default_factory=dict)

    outcome: DecisionOutcome = DecisionOutcome.APPROVED

    reason: str | None = None

    evaluated_at: datetime = field(default_factory=datetime.utcnow)